common/djangoapps/entitlements/models.py

```diff
@@ -24,3 +54,65 @@ class CourseEntitlement(TimeStampedModel):
         help_text='The current Course enrollment for this entitlement. If NULL the Learner has not enrolled.'
     )
     order_number = models.CharField(max_length=128, null=True)
+    _policy = models.ForeignKey(CourseEntitlementPolicy, null=True, blank=True)
+
+    @property
+    def expired_at_datetime(self):
+        if not self.expired_at and self.get_days_until_expiration() < 0:
+            self.expired_at = datetime.utcnow()
+            self.save()
+        return self.expired_at
+
+    @expired_at_datetime.setter
+    def expired_at_datetime(self, value):
+        self.expired_at = value
+
+    @property
+    def policy(self):
+        return self._policy or CourseEntitlementPolicy()
+
+    @policy.setter
+    def policy(self, value):
+        self._policy = value
+
+    def get_days_since_created(self):
+        """
+        Returns an integer of number of days since the entitlement has been created
+        """
+        utc = pytz.UTC
+        return (datetime.now(tz=utc) - self.created).days
+
+    def get_days_until_expiration(self):
```

Comment:
I'm a little concerned that this assumption is not always going to be a calculable field or could be misused by future devs.   An unused entitlement will expire in...450 days.  Then they enroll in a session and now this value is 14 days.

If anyone uses this to "predict" the future there could be issues.

That's not to mention the fact that we may include logic beyond number of days in these policies.  E.G. Never expire if the user is enrolled in a prerequisite course which is still active.

Reply:
Not sure I understand.  This is purposefully dynamic based on the policy and usage of the entitlement.